ParkingFinder/services/user.py

```diff
@@ -59,18 +61,47 @@ def register(user, vehicle=None):
             vehicle.validate()
 
         _user = yield UserRepository.upsert(user=user)
+        if vehicle:
+            try:
+                vehicle_exist = yield VehicleRepository.retrieve_vehicle_by_plate(vehicle.plate)
+            except NoResultFound:
+                VehicleRepository.insert_registered_vehicle(user_id=user.user_id, vehicle=vehicle)
 
         raise Return(_user)
 
-        # TODO if vehicle not None, add to vehicle repository
-
     @staticmethod
     @coroutine
     def get_user_detail(user_id):
+        """
+        Return the user information in detail with registered car information
+
+        :param String user_id:
+        :return: User user
+        """
         try:
             user = yield UserRepository.read_one(user_id=user_id)
-            # TODO read vehicles from vehicle repository and assign to user entity
+            user_vehicles = yield VehicleRepository.retrieve_vehicle_by_user(user_id=user_id)
+            user.owned_vehicles = user_vehicles
         except NoResultFound:
             raise NotFound
 
         raise Return(user)
+
+    @staticmethod
+    @coroutine
+    def activate_vehicle(user, vehicle):
+        """
+        activate an vehicle of a user and the vehicle that has been passed in
+        must have a register relationship with user by the front-end
+
+        :param User user:
+        :param Vehicle vehicle:
+        :return: updated user
+        """
+        user.validate()
+        vehicle.validate()
```

Comment:
We need to check whether the vehicle is in user.owned_vehicles list or not first. If it is, it is allowed to be the activate_vehicle.